soh/src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -298,6 +300,7 @@ void EnSth_GiveReward(EnSth* this, PlayState* play) {
         this->actor.parent = NULL;
         EnSth_SetupAction(this, EnSth_RewardObtainedTalk);
         gSaveContext.eventChkInf[EVENTCHKINF_SKULLTULA_REWARD_INDEX] |= this->eventFlag;
+        GameInteractor_ExecuteOnFlagSet(FLAG_EVENT_CHECK_INF, (EVENTCHKINF_SKULLTULA_REWARD_INDEX << 4) + (int)log2(this->eventFlag));
```

Comment:
another instance where i think it makes sense to have an array with the values we need instead of including math and casting to int